x/ecocredit/server/credit_type.go

```diff
@@ -0,0 +1,26 @@
+package server
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+	"strings"
+)
+
+func (s serverImpl) getCreditType(ctx sdk.Context, creditTypeName string) (*ecocredit.CreditType, error) {
+	creditTypes := s.getAllCreditTypes(ctx)
+	creditTypeName = strings.ToLower(creditTypeName)
+	creditTypeName = strings.TrimSpace(creditTypeName)
+	for _, creditType := range creditTypes {
+		if creditType.Type == creditTypeName {
```

Comment:
Can we guarantee that `creditType.Type` is normalized?

Reply:
it should - the validation function for the param also calls ToLower and removes whitespace. i just realized though i didnt update this part to the new function in utils so i'll swap that

Reply:
As far as I can see the validation function is read only, so it does normalise while doing the checks, but doesn't affect the param that actually gets stored. Do you agree?

Reply:
ahh right.. would it be better to normalize it here? my initial thought on this is that it would be more ideal to transform the string to a normalized form *and then* store it but not sure if theres a way to do that 🤔

Reply:
Well, there's three pathways to storing it:
- The default value, which should be normalised.
- Some genesis file + import genesis. You can either validate and throw an error, or just normalise quietly.
- Governance update. This one I guess you can only validate? I don't know which message is executed to update a param during governance proposal, so maybe check that and see if there's any chance of modifying the value that gets stored?

Reply:
i think there is still the concern if governance updates the params, and gives us something like CaRbOn - we don't really have access to normalizing that before storage? i believe that power lies inside the sdk. 

maybe it would be better to just check normalized string against the given string and if they dont equal, throw error? that way we have a hard enforcement on normalized strings?

Reply:
ok i strictly enforce normalizaiton at the validation level here: https://github.com/regen-network/regen-ledger/pull/428/commits/231edf12dba2bc5377d5140b1d9f5a24c07f2501

thoughts on this? @aaronc @ruhatch

Reply:
Yeah, I think that seems reasonable to me. At what stage in the proposal process is that validation function called? Does it check it before the proposal is submitted, because it would be an annoying UX if they vote a proposal through and then it fails with that kind of normalisation error!!

Reply:
it looks like it will be validated upon proposal request see: <https://github.com/cosmos/cosmos-sdk/blob/master/x/params/proposal_handler.go#L26>

which calls update which then calls validate, which then calls our validate funtion :^) so looks like the UX will be ok